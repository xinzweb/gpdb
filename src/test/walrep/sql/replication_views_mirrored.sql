-- Check how many WAL replication mirrors are up
SELECT count(*) FROM gp_segment_configuration where preferred_role='m' and status='u';

-- Check how many WAL replication mirrors are up and not-in-sync
-- FIXME: mode should be 's', pending FTS changes
SELECT count(*) FROM gp_segment_configuration where preferred_role='m' and mode='n';

-- Check pg_stat_replication view (this assumes standby master has not been created)
SELECT * FROM pg_stat_replication;

-- Check pg_stat_replication view of master and primary segments
SELECT gp_segment_id, application_name, state, sync_state FROM gp_stat_replication;
