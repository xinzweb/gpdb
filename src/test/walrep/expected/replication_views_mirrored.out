-- Check how many WAL replication mirrors are up
SELECT count(*) FROM gp_segment_configuration where preferred_role='m' and status='u';
 count
-------
     4
(1 row)

-- Check how many WAL replication mirrors are up and in-sync
SELECT count(*) FROM gp_segment_configuration where preferred_role='m' and mode='s';
 count 
-------
     4
(1 row)

-- Check pg_stat_replication view of master and primary segments
SELECT gp_segment_id, application_name, state, sync_state FROM gp_stat_replication;
 gp_segment_id | application_name |   state   | sync_state 
---------------+------------------+-----------+------------
             0 | walreceiver      | streaming | sync
             1 | walreceiver      | streaming | sync
             2 | walreceiver      | streaming | sync
            -1 | walreceiver      | streaming | sync
(4 rows)

